src/MountedTraversal.js

```diff
@@ -49,9 +49,8 @@ export function instHasClassName(inst, className) {
   if (!isDOMComponent(inst)) {
     return false;
   }
-  let classes = findDOMNode(inst).className || '';
-  classes = classes.replace(/\s/g, ' ');
-  return ` ${classes} `.indexOf(` ${className} `) > -1;
+  let classes = findDOMNode(inst).classList;
```

Comment:
That would cripple existing functionality in those browsers where `classList` is unsupported. enzyme needs to not just "not break" in older browsers, it needs to "work as perfectly as possible".